test/rosdep_repo_check/alpine.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2021, Open Source Robotics Foundation
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     * Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in the
+#       documentation and/or other materials provided with the distribution.
+#     * Neither the name of the Willow Garage, Inc. nor the names of its
+#       contributors may be used to endorse or promote products derived from
+#       this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import os
+import tarfile
+from io import TextIOWrapper
+
+from . import open_gz_url
+from . import PackageEntry
+from . import RepositoryCacheCollection
+
+
+def replace_tokens(string, os_code_name, os_arch):
+    """Replace tokens in the repository base URL."""
+    for key, value in {
+        '$basearch': os_arch,
+        '$releasever': os_code_name,
+    }.items():
+        string = string.replace(key, value)
+    return string
```

Comment:
Sorry, not sure what you're asking me to do here. Is this incorrect? I based it off the `rpm.py` script hence having the similar code. I imagined the style was more aligned with RPM than it was debian.

Reply:
What I'm trying to say is that the goal isn't to align the style with RPM or deb, but rather alpine. The RPM base URL tokens are the same as you see in `/etc/yum.repos.d/fedora.repo`, and the base URLs for the debian repositories are similar to what you find in `/etc/apt/sources.list`.

Here's what `/etc/apk/repositories` looks like:
```
https://dl-cdn.alpinelinux.org/alpine/edge/main
https://dl-cdn.alpinelinux.org/alpine/edge/community
```

It looks like we can't avoid using a token for the release part (in this case, `edge`) but it appears that the rest of the URL (the arch-specific part) follows a known structure. I'm OK with borrowing `$releasever` from RPM to make this work, but I'd rather not deviate from the "native" representation of an alpine base URL any more than we need to.

tl;dr, I'd like to see this:
```
!alpine_base_url https://dl-cdn.alpinelinux.org/alpine/$releasever/main
```
...rather than this:
```
!alpine_base_url https://dl-cdn.alpinelinux.org/alpine/$releasever/main/$basearch
```

Reply:
Understood now. Fixed.